app/controllers/staff/training_sessions_controller.rb

```diff
@@ -0,0 +1,130 @@
+class Staff::TrainingSessionsController < StaffAreaController
+  before_action :load_user, only: [:show, :edit, :update]
+
+  layout 'staff_area'
+
+  def index
+    if sort_params
+      if params[:p] == "signed_in_users"
+        if !params[:sort].present? && !params[:direction].present?
+          params[:sort] = "lab_sessions.sign_in_time"
+          params[:direction] = "desc"
+        end
+        @users_temp = LabSession.joins(:user).where("sign_out_time > ?", Time.now)
+        if params[:location].present?
+          @users_temp = @users_temp.joins("INNER JOIN pi_readers ON pi_mac_address = mac_address AND LOWER(pi_location) = LOWER('#{params[:location]}')")
+        end
+        @users_temp = @users_temp.order("#{params[:sort]} #{params[:direction]}").paginate(:page => params[:page], :per_page => 20)
+        @users = @users_temp.includes(:user).map{|session| session.user}
+        @total_pages = @users_temp.total_pages
+      elsif params[:p] == "new_users" || !params[:p].present?
+        if !params[:sort].present? && !params[:direction].present?
+          params[:sort] = "users.created_at"
+          params[:direction] = "desc"
+        end
+        @users = User.includes(:lab_sessions).order("#{params[:sort]} #{params[:direction]}").paginate(:page => params[:page], :per_page => 20)
+        @total_pages = @users.total_pages
+      end
+    else
+      redirect_to staff_index_path
+      flash[:alert] = "Invalid parameters!"
+    end
+  end
+
+  def sort_params
+    if (((["username","name","lab_sessions.sign_in_time","users.created_at"].include? params[:sort]) && (["desc","asc"].include? params[:direction])) || (!params[:sort].present? && !params[:direction].present?))
+      return true
+    end
+  end
+
+  def new
+    training_session = TrainingSession.new(training_session_params)
+    if params['training_session_users'].present?
+      params['training_session_users'].each do |user|
+        training_session.users << User.find(user)
+      end
+    end
+    redirect_to :back
+  end
+
+  def create
+    staff = current_user
+    training_session = TrainingSession.new(training_session_params)
+    if training_session.save
+      flash[:notice] = "Training session created succesfully"
+    end
+    redirect_to staff_training_sessions_url
+  end
+
+  def rename_training_session
+    staff = current_user
+    training_session = TrainingSession.find_by(training_session_params)
+    if params['training_session_new_training_id'].present?
+      training_session.training_id = params['training_session_new_training_id']
+      if training_session.save
+        flash[:notice] = "Training session renamed succesfully"
+      end
+    end
+    redirect_to :back
+  end
+
+  def reschedule_training_session
+    staff = current_user
+    training_session = TrainingSession.find_by(training_session_params)
```

Comment:
I wrote up a longer message in slack, but instead of sending the 3 parameters to figure out which training session to update, you should use the ID of the training_session once it has been created. Once you pass in the ID you can just do `TraingSession.find(params[:id])` which is how the the admin/users_controller works